Indexer/src/main/java/Indexer/Indexer.java

```diff
@@ -48,19 +58,31 @@ public void run() {
         }
     }
 
-    private BulkResponse index(ConsumerRecords<String, Map> records) {
-
+    private void index(ConsumerRecords<String, String> records) {
+        ObjectMapper mapper = new ObjectMapper();
         BulkRequest elasticBulkRequest = new BulkRequest();
 
-        try {
-            for (ConsumerRecord<String, Map> record : records) {
-                elasticBulkRequest.add(new IndexRequest(index, logType).source(record.value()));
+        for (ConsumerRecord<String, String> record : records) {
+            try{
+                KafkaRecord kafkaRecord = mapper.readValue(record.value(), KafkaRecord.class);
+                Optional<AccountData> OptionalAccountData = accountsClient.getAccount(kafkaRecord.getAccountToken());
+                AccountData accountData = OptionalAccountData.get();
```

Comment:
You assume here that the account is present - this could be a tricky assumption - the account might be deleted while still have logs in kafka